openedx/core/lib/block_structure/transformers.py

```diff
@@ -105,3 +110,40 @@ def is_collected_outdated(cls, block_structure):
             )
 
         return bool(outdated_transformers)
+
+    def _transform_optimized(self, transformers, block_structure):
+        """
+        Transforms the given block_structure using the transform_block_filter
+        method from the given transformers.
+        """
+        if not transformers:
+            return
+
+        removal_filters = []
```

Comment:
The `transform_block_filter` method is generic and can be anything.  It's essentially really a `map_with_filter` function.  So can we rename it so it's not specific to removal?